### PR TITLE
Added size() to BoundedSourceQueue #30124

### PR DESCRIPTION
**Pull request by azotcsit**
References #30124

Please, check the reference issue for more details about below changes.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/BoundedSourceQueue.scala [hunk 1]
```diff
@@ -32,4 +32,11 @@ trait BoundedSourceQueue[T] {
    * Completes the stream with a failure.
    */
   def fail(ex: Throwable): Unit
+
+  /**
+   * Returns the approximate number of elements in this queue.
+   *
+   * @return the approximate number of elements in this queue
+   */
+  def size(): Integer
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/BoundedSourceQueue.scala:41]: use `Int` to avoid boxing

Comment (alex-ninja) [reply akka-stream/src/main/scala/akka/stream/BoundedSourceQueue.scala]: Done

File: akka-stream/src/main/scala/akka/stream/BoundedSourceQueue.scala [hunk 2]
```diff
@@ -32,4 +32,11 @@ trait BoundedSourceQueue[T] {
    * Completes the stream with a failure.
    */
   def fail(ex: Throwable): Unit
+
+  /**
+   * Returns the approximate number of elements in this queue.
+   *
+   * @return the approximate number of elements in this queue
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/BoundedSourceQueue.scala:39]: Saying it one time is enough ;) 

```suggestion
```

**Review by johanandren — APPROVED** [review]
LGTM (I don't think the name needs to be more explicit about the aproximateness)

**Comment by johanandren** [comment]
OK TO TEST

**Review by octonato — APPROVED** [review]

**Pull request merged**